### ISSUE TITLE
[P2] 指数関数と対数関数の逆関係・対数計算法を視覚化するデモページを実装する

### ISSUE THREAD

**Issue by tomiya7688** [tomiya7688/comfyUI_support_tools#95]
## 目的
数学的シミュレーション／数学デモの一部として、**指数関数と対数関数の関係**を、グラフ・数式・数値操作を連動させて視覚的に理解できるデモページを実装する。

特に対数については、単に `log` のグラフを表示するだけでなく、以下の「数学として独特な操作」を実際に触って確認できることを重視する。

- 底を変える
- 真数を変える
- 指数との逆関係を確認する
- 積・商・べきの対数法則を数値で確かめる
- 底の変換公式を確認する
- 定義域・底の制約を操作時に理解できる

---

## ページ構成
1ページ内を主に4セクションへ分ける。

### 1. 指数関数デモ
例:
```text
y = b^x
```

実装:
- [ ] 底 `b` をsliderで変更できる
- [ ] `x` をslider / dragで変更できる
- [ ] `y = b^x` のグラフを表示
- [ ] 現在点 `(x, b^x)` を表示
- [ ] `b > 1` と `0 < b < 1` で増加/減少が切り替わることを確認できる
- [ ] `b = 1` は定数関数になることを補足
- [ ] `b <= 0` を実数範囲の基本デモでは許可しない

視覚説明:
- 「指数関数は、入力xを指数として使う関数」
- 「底が1より大きいと増加、0と1の間なら減少」

---

### 2. 指数関数と対数関数の逆関係
例:
```text
b^x = y  <=>  log_b(y) = x
```

実装:
- [ ] `y = b^x` と `y = log_b(x)` を同一座標上に表示
- [ ] `y = x` の直線を補助表示
- [ ] 2つのグラフが `y=x` に関して対称になることを視覚化
- [ ] 指数側の点 `(x, y)` と対数側の点 `(y, x)` を連動表示
- [ ] 底 `b` を変更すると両グラフが同時更新
- [ ] 指数 `x` を変更すると対応する対数式も即時更新

表示例:
```text
2^3 = 8
log_2(8) = 3
```

視覚説明:
- 「対数は『何乗したらこの値になるか』を求める操作」
- 「指数関数と対数関数は互いに逆関数」

---

### 3. 対数の計算操作デモ
対数の計算法を、数式だけでなく数値操作で確認できるようにする。

#### 積の法則
```text
log_b(xy) = log_b(x) + log_b(y)
```

- [ ] `x`, `y`, `b` を操作可能
- [ ] 左辺と右辺を別々に数値計算して表示
- [ ] 同じ値になることをリアルタイム確認

#### 商の法則
```text
log_b(x/y) = log_b(x) - log_b(y)
```

- [ ] `x`, `y`, `b` を操作可能
- [ ] 左辺 / 右辺の数値を並べて比較

#### べきの法則
```text
log_b(x^k) = k log_b(x)
```

- [ ] `x`, `k`, `b` を操作可能
- [ ] 左辺 / 右辺を比較

#### 底の変換公式
```text
log_b(x) = log_c(x) / log_c(b)
```

- [ ] `b`, `c`, `x` を操作可能
- [ ] 同じ値になることを表示
- [ ] `c=10` / `c=e` のpresetを用意

重要:
- [ ] `x > 0` を満たさない入力は対数が実数で定義されないことをUI上で示す
- [ ] 底は `b > 0`, `b != 1` の制約をUI上で示す
- [ ] 不正値は単にNaN表示にせず、なぜ計算できないか短く説明する

---

### 4. 対数の「尺度変換」デモ
対数が大きな値の差を圧縮して表現する性質も、視覚的に確認できるようにする。

実装:
- [ ] 線形軸と対数軸を切り替えて同じ値列を表示
- [ ] `1, 10, 100, 1000` 等が対数軸で等間隔に並ぶことを表示
- [ ] `log10(1)=0`, `log10(10)=1`, `log10(100)=2` を表示
- [ ] 「桁数・倍率を見る尺度」として簡単に説明

視覚説明:
- 「対数は足し算ではなく倍率の差を見やすくする」
- 「10倍ごとに1増える、という見方ができる」

---

## 共通UI
- [ ] グラフ・数式・数値の3表示を連動
- [ ] slider / dragで値を変えると即時更新
- [ ] 現在操作している変数を数式上で強調表示
- [ ] Resetを用意
- [ ] presetを用意

候補preset:
- base 2
- base 10
- base e
- `0 < b < 1` の減少例

---

## AIデモとの接続
ページ末尾に短い補足を入れる。

例:
- log loss / cross entropyで対数が使われる
- softmaxとlogの組合せ
- 掛け算の積をlogで足し算へ変換できる
- 非常に小さい確率の積を扱いやすくする

ただしこのIssueではAI応用の実装は行わず、数学的意味の補足に留める。

関連:
- #92 機械学習の学習ループ
- #94 微分・積分・偏微分・連鎖律

---

## テスト
- [ ] `2^3 = 8` と `log_2(8)=3` が連動表示される
- [ ] `y=b^x` と `y=log_b(x)` が `y=x` に対して逆関数関係になる
- [ ] `b>1` で指数関数増加、`0<b<1` で減少する
- [ ] 積の法則の左右辺が一致する
- [ ] 商の法則の左右辺が一致する
- [ ] べきの法則の左右辺が一致する
- [ ] 底の変換公式の左右辺が一致する
- [ ] `x<=0` の対数入力を適切に拒否・説明する
- [ ] `b<=0` または `b=1` を適切に拒否・説明する
- [ ] 線形軸 / 対数軸切替が正しく動く

---

## 完了条件
- 指数関数と対数関数が逆関数であることを視覚的に確認できる
- 底を操作すると指数・対数の両グラフが連動して変化する
- 対数の積・商・べき・底変換の各法則を数値操作で確認できる
- 対数の定義域と底の制約を操作中に理解できる
- 対数軸で「倍率を扱う尺度」であることを視覚的に確認できる
- 数式を読むだけでなく、触って対数計算の特殊性を理解できるページになっている

## Out of Scope
- 複素対数
- リーマン面
- 任意精度計算
- 記号計算CAS
- log loss / cross entropyの本格AIデモ
- 指数分布・対数正規分布などの確率分布デモ
